src/python/pants/core/util_rules/adhoc_process_support_test.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2024 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+from __future__ import annotations
+
+import os
+
+import pytest
+
+from pants.core.util_rules.adhoc_process_support import _path_metadata_to_bytes
+from pants.engine.fs import PathMetadataRequest, PathMetadataResult
+from pants.engine.internals.native_engine import PathMetadata
+from pants.engine.rules import QueryRule
+from pants.testutil.rule_runner import RuleRunner
+
+
+@pytest.fixture
+def rule_runner() -> RuleRunner:
+    return RuleRunner(
+        rules=[
+            QueryRule(PathMetadataResult, (PathMetadataRequest,)),
+        ],
+        isolated_local_store=True,
+    )
+
+
+def test_path_metadata_to_bytes(rule_runner: RuleRunner) -> None:
```

Comment:
Maybe we could also have tests that catch the `struct.pack` problem? in both ways:

1. verify that passing the same metadata gives the same result (or different in irrelevant ways, like access time)
2. verify that passing different metadata (even only slightly different, like different mtime) gives different results

Reply:
Both suggestions sound like good tests. Will add.

Reply:
Added.